pkg/scm/gitlab/context.go

```diff
@@ -158,3 +164,49 @@ func (c *Context) HasExecutedActionGroup(group string) bool {
 
 	return ok
 }
+
+// GetCodeOwners returns the eligible code owners for the merge request
+//
+// This is based on the elibible approvers in the rules of the merge request approval
+// state api.
+func (c *Context) GetCodeOwners() scm.Actors {
+	actors := make(scm.Actors, 0)
+
+	for _, rule := range c.MergeRequest.ApprovalState.Rules {
+		// Multiple code owner paths could be matched when sections are used, so
+		// flatten the list of eligible approvers
+		if *rule.Type != ApprovalRuleTypeCodeOwner {
```

Comment:
I wonder in what case `rule.Type` can be `nil`, since the GraphQL API has it marked as nill-able; maybe worth guarding against just-in-case :)